tests/integration/test_jenkins.py

```diff
@@ -98,3 +100,46 @@ async def test_jenkins_automatic_update(
 
     updated_workload_version = updated_app_status.workload_version
     assert updated_workload_version == latest_jenkins_lts_version, "The Jenkins should be updated."
+
+
+async def test_jenkins_persist_jobs_on_restart(
+    model_app_unit: ModelAppUnit,
+    jenkins_client: jenkinsapi.jenkins.Jenkins,
+    jenkins_new_job_configuration: str,
+):
+    """
+    arrange: a bare Jenkins charm.
```

Comment:
It looks simpler and lees error prone to:
- Deploy Jenkins
- Check the content of the config file don't match ` jenkins_new_job_configuration`
- Reboot with Unit.reboot() https://github.com/canonical/operator/releases/tag/2.8.0
- - Check the content of the config file match ` jenkins_new_job_configuration`

Reply:
Thanks, I updated the test to reflect this